optuna_dashboard/_app.py

```diff
@@ -449,6 +452,46 @@ def save_trial_note(study_id: int, trial_id: int) -> dict[str, Any]:
         response.status = 204  # No content
         return {}
 
+    @app.get("/csv/<study_id:int>")
+    def download_csv(study_id: int) -> BottleViewReturn:
+        # Create a CSV file
+        summary = get_study_summary(storage, study_id)
+        if summary is None:
+            response.status = 404  # Not found
+            return {"reason": f"study_id={study_id} is not found"}
+        trials = get_trials(storage, study_id)
+
+        param_names = sorted(set(chain.from_iterable([t.params.keys() for t in trials])))
+        user_attr_names = sorted(set(chain.from_iterable([t.user_attrs.keys() for t in trials])))
+
+        param_names_header = [f"Param {x}" for x in param_names]
+        user_attr_names_header = [f"UserAttribute {x}" for x in user_attr_names]
+        value_header = ["Value"]
+        n_objs = max([len(t.values) for t in trials if t.values is not None], default=1)
+        if n_objs > 1:
+            value_header = [f"Objective {x}" for x in range(n_objs)]
```

Comment:
I modified so that we can take the metric_names.

```suggestion
        try:
            study_name = storage.get_study_name_from_id(study_id)
            study = optuna.load_study(storage=storage, study_name=study_name)
        except KeyError:
            response.status = 404  # Not found
            return {"reason": f"study_id={study_id} is not found"}

        trials = study.trials
        param_names = sorted(set(chain.from_iterable([t.params.keys() for t in trials])))
        user_attr_names = sorted(set(chain.from_iterable([t.user_attrs.keys() for t in trials])))

        param_names_header = [f"Param {x}" for x in param_names]
        user_attr_names_header = [f"UserAttribute {x}" for x in user_attr_names]
        n_objs = len(study.directions)
        if study.metric_names is not None:
            value_header = study.metric_names
        elif n_objs == 1:
            value_header = ["Value"]
        else:
            value_header = [f"Objective {x}" for x in range(n_objs)]
```

Reply:
IMO, it would be better to avoid using the private attribute `_system_attrs`. How about loading the study as @nabenabe0928 suggested?

Reply:
OK. I followed the suggestion, and stopped to use `get_study_summary` in `download_csv` .